### PR TITLE
Code style

### PR DESCRIPTION
**Pull request by Henni**
TODOs:
- [x] add eslint
- [x] run eslint and test on travis
- [x] fix code style
- [x] add .editorconfig

### CODE REVIEW

**Comment by Henni** [comment]
diff without whitespace changes: https://github.com/MusicConnectionMachine/api/pull/58/files?w=1

File: app.js [hunk 1]
```diff
@@ -17,7 +13,7 @@ var config = {
 };
 
 
-require(path.join(__dirname, "index.js")).connect(function (context) {
+require(path.join('./index.js')).connect(function (context) {
```
Comment (felixschorer) [inline app.js]: `path.join()` isn't required if you're actually not joining paths.

File: package.json [hunk 2]
```diff
@@ -15,12 +15,16 @@
...
+    "npm-run-all": "^4.0.2",
     "should": "^7.1.0",
     "supertest": "^1.0.0"
   },
   "scripts": {
     "postinstall": "npm install -g multi-file-swagger",
     "start": "cd ./api/swagger/src/ && multi-file-swagger ./swagger.yaml > ../swagger.json && cd ../../../ && node app.js",
-    "test": "swagger project test"
+    "test": "npm-run-all test:swagger lint",
+    "test:swagger": "swagger_swagger_fileName=api/swagger/src/swagger.yaml swagger project test",
```
Comment (felixschorer) [inline package.json]: Does this work? Shouldn't it use the compiled `swagger.json` instead? Last time I've check the `swagger.yaml` file was split up into multiple smaller files needed to be compiled into one big `.json` for actually running the API.

File: app.js (hunk 1 above)
Comment (Sandr0x00) [reply app.js]: the `__dirname` stuff is required here, otherwise you get `Error: Cannot find module 'index.js'`

File: package.json (hunk 2 above)
Comment (Henni) [reply package.json]: I'm not sure myself. But the previous command just failed immediately and swagger.json isn't generated on travis. So I switched to a the yaml file which at least doesn't throw any errors.

I'll take a closer look at this tomorrow. But it doesn't break anything so let's merge this and change it in another PR.

File: app.js (hunk 1 above)
Comment (Henni) [reply app.js]: Thanks, I'll look into it.

Comment (Henni) [reply app.js]: @Sandr00 I can't reproduce your issue. Both `npm start` and `npm test` run without an issue for me.

File: package.json (hunk 2 above)
Comment (felixschorer) [reply package.json]:
```
$ cd ./api/swagger/src/
$ multi-file-swagger ./swagger.yaml > ../swagger.json
$ cd ../../../
``` 
will compile the `swagger.yaml` file to `api/swagger/swagger.json`. The directory changes are actually necessary as the compiler tool will fail when it isn't being run in the same directory as the `swagger.yaml` file.

> But it doesn't break anything so let's merge this and change it in another PR.

It's one of the main points of this PR, so it should be working before merging in my opinion.

Comment (Henni) [reply package.json]: @lgfrbcsgo thanks for the commands. Now I finally understand what multi-file-swagger does.

This should now be fixed in 6ae92d9

**Review by felixschorer — APPROVED** [review]
Nicely done! 👍

File: app.js (hunk 1 above)
Comment (Sandr0x00) [reply app.js]: Got it with `docker-compose up`

**Pull request merged**